internal/hcsoci/layers.go

```diff
@@ -192,6 +184,33 @@ func MountContainerLayers(ctx context.Context, layerFolders []string, guestRoot
 	return rootfs, nil
 }
 
+func addLCOWLayer(ctx context.Context, uvm *uvmpkg.UtilityVM, layerPath string) (uvmPath string, err error) {
+	// don't try to add as vpmem when we want the uvm to be fully physically backed
+	if !uvm.FullyPhysicallyBacked() {
+		// We first try vPMEM and if it is full or the file is too large we
+		// fall back to SCSI.
+		uvmPath, err = uvm.AddVPMEM(ctx, layerPath)
+		if err == nil {
+			return uvmPath, err
+		}
+	}
+
+	if err == uvmpkg.ErrNoAvailableLocation || err == uvmpkg.ErrMaxVPMEMLayerSize {
```

Comment:
```
if err != nil && err != uvmpkg.ErrNoAvailableLocation && err != uvmpkg.ErrMaxVPMEMLayerSize {
    return "", err
}
```

Reply:
Actually, we should move the error handling for the `AddVPMEM` case up into the if block above.